src/test/scala/io/iohk/ethereum/jsonrpc/EthJsonRpcControllerSpec.scala

```diff
@@ -0,0 +1,228 @@
+package io.iohk.ethereum.jsonrpc
+
+import io.iohk.ethereum.Fixtures
+import io.iohk.ethereum.db.components.{SharedEphemDataSources, Storages}
+import io.iohk.ethereum.domain.{Block, BlockchainImpl}
+import io.iohk.ethereum.jsonrpc.JsonSerializers.{BlockResponseSerializer, QuantitiesSerializer}
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import org.json4s.{DefaultFormats, Extraction}
+import org.json4s.JsonAST._
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatest.prop.PropertyChecks
+import org.spongycastle.util.encoders.Hex
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+class EthJsonRpcControllerSpec extends WordSpec with Matchers with PropertyChecks {
```

Comment:
imo we need to test 3 things:

1. XXXService returns expected data and handle corner cases and errors if needed
2. We decode/encode requests & responses properly
3. The service is handled by the routes

What do you think? Also, I'm  not sure if the cases you mention, @rtkaczyk , cover this.